File: app/src/main/java/org/ole/planet/myplanet/ui/team/teamCourse/TeamCourseFragment.java
package org.ole.planet.myplanet.ui.team.teamCourse;

import android.os.Bundle;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;

import androidx.annotation.NonNull;
import androidx.annotation.Nullable;
import androidx.recyclerview.widget.LinearLayoutManager;

import org.ole.planet.myplanet.databinding.FragmentTeamCourseBinding;
import org.ole.planet.myplanet.model.RealmMyCourse;
import org.ole.planet.myplanet.ui.team.BaseTeamFragment;

import io.realm.RealmResults;

public class TeamCourseFragment extends BaseTeamFragment {
    private FragmentTeamCourseBinding fragmentTeamCourseBinding;

    public TeamCourseFragment() {
    }

    @Override
    public View onCreateView(@NonNull LayoutInflater inflater, ViewGroup container, Bundle savedInstanceState) {
        fragmentTeamCourseBinding = FragmentTeamCourseBinding.inflate(inflater, container, false);
        return fragmentTeamCourseBinding.getRoot();
    }

    @Override
    public void onActivityCreated(@Nullable Bundle savedInstanceState) {
        super.onActivityCreated(savedInstanceState);
        RealmResults<RealmMyCourse> courses = mRealm.where(RealmMyCourse.class).in("id", team.courses.toArray(new String[0])).findAll();
        AdapterTeamCourse adapterTeamCourse = new AdapterTeamCourse(getActivity(), courses, mRealm, teamId, settings);
        fragmentTeamCourseBinding.rvCourse.setLayoutManager(new LinearLayoutManager(getActivity()));
        fragmentTeamCourseBinding.rvCourse.setAdapter(adapterTeamCourse);
        showNoData(fragmentTeamCourseBinding.tvNodata, adapterTeamCourse.getItemCount());
    }
}

File: app/src/main/java/org/ole/planet/myplanet/ui/team/teamCourse/TeamCourseFragment.kt
package org.ole.planet.myplanet.ui.team.teamCourse

import android.os.Bundle
import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import androidx.lifecycle.lifecycleScope
import androidx.recyclerview.widget.LinearLayoutManager
import dagger.hilt.android.AndroidEntryPoint
import kotlinx.coroutines.launch
import org.ole.planet.myplanet.databinding.FragmentTeamCourseBinding
import org.ole.planet.myplanet.model.RealmNews
import org.ole.planet.myplanet.ui.team.BaseTeamFragment

@AndroidEntryPoint
class TeamCourseFragment : BaseTeamFragment() {
    private var _binding: FragmentTeamCourseBinding? = null
    private val binding get() = _binding!!
    private var adapterTeamCourse: AdapterTeamCourse? = null

    override fun onCreateView(inflater: LayoutInflater, container: ViewGroup?, savedInstanceState: Bundle?): View {
        _binding = FragmentTeamCourseBinding.inflate(inflater, container, false)
        return binding.root
    }

    override fun onViewCreated(view: View, savedInstanceState: Bundle?) {
        super.onViewCreated(view, savedInstanceState)
        setupCoursesList()
    }

    private fun setupCoursesList() {
        binding.rvCourse.layoutManager = LinearLayoutManager(activity)
        team?.let { currentTeam ->
            val courseIds = currentTeam.courses?.toList() ?: emptyList()
            if (courseIds.isEmpty()) {
                showNoData(binding.tvNodata, 0, "teamCourses")
                return@let
            }

            viewLifecycleOwner.lifecycleScope.launch {
                val courses = courseRepository.getCoursesByTeamCourseIds(courseIds)
                val teamCreator = currentTeam.createdBy ?: ""
                adapterTeamCourse = settings?.let {
                    AdapterTeamCourse(requireActivity(), courses, teamCreator, it)
                }
                binding.rvCourse.adapter = adapterTeamCourse
                adapterTeamCourse?.let {
                    showNoData(binding.tvNodata, it.itemCount, "teamCourses")
                }
            }
        } ?: showNoData(binding.tvNodata, 0, "teamCourses")
    }
    override fun onNewsItemClick(news: RealmNews?) {}
    override fun clearImages() {
        imageList.clear()
        llImage?.removeAllViews()
    }

    override fun onDestroyView() {
        _binding = null
        super.onDestroyView()
    }
}
